### PR TITLE
Suite comp

### PR DESCRIPTION
**Pull request by flore77**
## Need

```gherkin
As a developer
I need a suite component
So that I can display the test suites
```

## Deliverables

A component that will render the suites with their tests.

## Solution

The `Results` component will render the `Suite` component, more precise it will render an entire `array` of `Suites` and will pass via props to each element the suite from the data array.

The `Suite` will iterate over its tests and will build an array with `Test elements`, each test will receive as `key` the array `index`, i.e the position in the test array of each suite. There will be no conflicts, because they are not nested suites, i.e they are not brothers.

The `whole suite block` will be indented: `MARGIN x (suite.indent - 1) px`. We subtract 1, because only the global suite has 0, but we want that our "global" suites are indented at level 0.

The `suite title font-size` will be set conform: `100 / indent.level %`. We doesn't subtract 1, because then there will be division by 0, which is NaN.

Tests:

* should indent the suite properly
* should render the suite title
* should set the suite title fontSize correctly
* should render the tests array

### CODE REVIEW

**Comment by flore77** [comment]
@andrei-picus-hs please review specs

**Comment by andrei-picus-hs** [comment]
> The whole suite block will be indented: MARGIN x (suite.indent - 1) px

You can set the value on the component's [statics](http://devdocs.io/react/docs/component-specs#statics). You can then later access it in tests by looking at the component class i.e. `Component.MARGIN`, and in code by using `this.constructor.MARGIN`.

LGTM.

**Comment by flore77** [comment]
sure. I thought to make a `constant` folder and put them there in separate files, but this is better!!

**Comment by flore77** [comment]
@andrei-picus-hs please review

File: tests/unit/components/suite.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  beforeEach(function() {
+    tree = sd.shallowRender(<Suite suite={fixture}/>);
+  });
+
+  it('should indent the suite properly', function() {
+    var expected = {
+      marginLeft: Suite.MARGIN_LEFT * (fixture.indent - 1) + 'px'
+    };
+
+    expect(tree.findNode('.suite').props.style).to.be.deep.equal(expected);
```
Comment (andrei-picus-hs) [inline tests/unit/components/suite.js]: Compare just the `marginLeft` property. The test should be as focused as possible. If in the future you add another style property then this test will fail without a reasonable meaning.

File: tests/unit/components/suite.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  it('should render the suite title', function() {
+    expect(tree.textIn('.suite-title')).to.be.equal(fixture.title);
+  });
+
+  it('should set the suite title fontSize correctly', function() {
+    var expected = {
+      fontSize: Suite.FONT_SIZE / fixture.indent + '%'
+    };
+
+    expect(tree.findNode('.suite-title').props.style).to.deep.equal(expected);
```
Comment (andrei-picus-hs) [inline tests/unit/components/suite.js]: Same here.

File: ui/components/suite.jsx [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
 var Suite = React.createClass({
+  statics: {
+    MARGIN_LEFT: 20,
+    FONT_SIZE: 100
+  },
   render: function() {
-    return null;
+    var suite = this.props.suite,
+        suiteStyle = {
+          marginLeft: this.constructor.MARGIN_LEFT * (suite.indent - 1) + 'px'
```
Comment (andrei-picus-hs) [inline ui/components/suite.jsx]: You don't need to add `px`. React takes care of that for you.

**Comment by andrei-picus-hs** [comment]
Finished review. @flore77 needs pics!

File: ui/components/suite.jsx (hunk 3 above)
Comment (flore77) [inline ui/components/suite.jsx]: Not really...the test is failing :cry:

**Comment by flore77** [comment]
image

I have changed the mounted node to `document.body`, else I get the following error:

image

File: ui/components/suite.jsx (hunk 3 above)
Comment (flore77) [inline ui/components/suite.jsx]: image  neither in the browser

Comment (andrei-picus-hs) [inline ui/components/suite.jsx]: Look closely at the test: 'Expected `680` (without px) to equal `680px` (with px)'. You need to update the test and stop adding `px` there too :smile:

**Comment by flore77** [comment]
@andrei-picus-hs please review

File: tests/unit/components/suite.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+describe('Suite', function() {
+  var tree;
+
+  beforeEach(function() {
+    tree = sd.shallowRender(<Suite suite={fixture}/>);
+  });
+
+  it('should indent the suite properly', function() {
+    var expected = {
+      marginLeft: Suite.MARGIN_LEFT * (fixture.indent - 1)
```
Comment (andrei-picus-hs) [inline tests/unit/components/suite.js]: You can get rid of the object and put the value directly in the `expected` var.

File: tests/unit/components/suite.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    expect(tree.findNode('.suite').props.style.marginLeft).to.be.deep.
```
Comment (andrei-picus-hs) [inline tests/unit/components/suite.js]: See above + no need for deep comparison.

File: tests/unit/components/suite.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  it('should render the suite title', function() {
+    expect(tree.textIn('.suite-title')).to.be.equal(fixture.title);
+  });
+
+  it('should set the suite title fontSize correctly', function() {
+    var expected = {
+      fontSize: Suite.FONT_SIZE / fixture.indent + '%'
+    };
+
+    expect(tree.findNode('.suite-title').props.style.fontSize).to.deep.
```
Comment (andrei-picus-hs) [inline tests/unit/components/suite.js]: See above.

File: ui/components/suite.jsx [hunk 7]
```diff
@@ -1,8 +1,29 @@
...
+        },
+        tests;
+
+    tests = suite.tests.map(function(test, index) {
+      return <Test test={test} key={index}/>
+    });
+
+    return <div className="suite" style={suiteStyle}>
+      <h2 className="suite-title" style={titleStyle}>{suite.title}</h2>
+      {tests}
```
Comment (andrei-picus-hs) [inline ui/components/suite.jsx]: Put the JS code directly here instead of using a variable to store it.

**Comment by andrei-picus-hs** [comment]
Finished review.

**Comment by flore77** [comment]
@andrei-picus-hs please review

File: ui/components/suite.jsx [hunk 8]
```diff
@@ -1,8 +1,28 @@
...
+          marginLeft: this.constructor.MARGIN_LEFT * (suite.indent - 1)
+        },
+        titleStyle = {
+          fontSize: this.constructor.FONT_SIZE / suite.indent + '%'
+        };
+
+    return <div className="suite" style={suiteStyle}>
+      <h2 className="suite-title" style={titleStyle}>{suite.title}</h2>
+      {
+        suite.tests.map(function(test, index) {
```
Comment (andrei-picus-hs) [inline ui/components/suite.jsx]: Put this on the same as line as the opening curly bracket.

File: ui/components/suite.jsx [hunk 9]
```diff
@@ -1,8 +1,28 @@
...
+          return <Test test={test} key={index}/>
+        })
```
Comment (andrei-picus-hs) [inline ui/components/suite.jsx]: And this on the same as line as the closing curly bracket.

**Comment by andrei-picus-hs** [comment]
LGTM

**Pull request merged**